libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -957,6 +957,23 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			);
 			break;
 		}
+		case FunctionType::Kind::Wrap:
+		case FunctionType::Kind::Unwrap:
+		{
+			solAssert(arguments.size() == 1, "");
+			Type const* argumentType = arguments.at(0)->annotation().type;
+			solAssert(argumentType, "");
+			FunctionType::Kind kind = functionType->kind();
+			auto userDefinedType = kind == FunctionType::Kind::Wrap ?
```

Comment:
Wouldn't it be easier and less confusing to retrieve this from `_functionCall.expression`? Maybe not.

Reply:
On that note test cases for stuff like ``(MyType).wrap`` and the like would be good.